### PR TITLE
thin: add examples

### PR DESCRIPTION
**Pull request by e-marshall**
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by dcherian** [comment]
Thanks @e-marshall 

The doctest failure is because you need to add a few spaces to align `17 ...` (I clicked "details" to see this).
image

**Comment by e-marshall** [comment]
> Thanks @e-marshall
> 
> The doctest failure is because you need to delete a few spaces to align `17 ...` (I clicked "details" to see this). image

Ok, thanks @dcherian , I'll make those changes

File: xarray/core/dataarray.py [hunk 1]
```diff
@@ -1474,6 +1474,41 @@ def thin(
         """Return a new DataArray whose data is given by each `n` value
         along the specified dimension(s).
 
+        Examples
+        ----------
```
Comment (dcherian) [inline xarray/core/dataarray.py:1478]:
```suggestion
        --------
```

Deleting these extra `-` should prevent a warning

File: xarray/core/dataset.py [hunk 2]
```diff
@@ -2615,6 +2615,42 @@ def thin(
             The keyword arguments form of ``indexers``.
             One of indexers or indexers_kwargs must be provided.
 
+        Examples
+        ----------
```
Comment (dcherian) [inline xarray/core/dataset.py:2619]:
```suggestion
        Examples
        --------
```

File: xarray/core/dataarray.py [hunk 3]
```diff
@@ -1474,6 +1474,41 @@ def thin(
         """Return a new DataArray whose data is given by each `n` value
         along the specified dimension(s).
 
+        Examples
```
Comment (dcherian) [inline xarray/core/dataarray.py:1477]: Is the one line description above clear enough?

Comment (e-marshall) [reply xarray/core/dataarray.py:1477]: Yes!

File: xarray/core/dataarray.py [hunk 4]
```diff
@@ -1472,13 +1472,45 @@ def thin(
...
-        DataArray.head
-        DataArray.tail
+            along the specified dimension(s).
+
+            Examples
+            --------
+            >>> x_arr = np.arange(0, 26)
+            >>> x_arr
+            array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
+        17, 18, 19, 20, 21, 22, 23, 24, 25])
```
Comment (dcherian) [inline xarray/core/dataarray.py:1482]:
```suggestion
            array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
                   17, 18, 19, 20, 21, 22, 23, 24, 25])
```

File: xarray/core/dataset.py [hunk 5]
```diff
@@ -2615,6 +2615,42 @@ def thin(
...
+        >>> x_arr = np.arange(0, 26)
+        >>> x_arr
+        array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
+        17, 18, 19, 20, 21, 22, 23, 24, 25])
```
Comment (dcherian) [inline xarray/core/dataset.py:2623]:
```suggestion
        array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
               17, 18, 19, 20, 21, 22, 23, 24, 25])
```

File: xarray/core/dataarray.py [hunk 6]
```diff
@@ -1472,13 +1472,45 @@ def thin(
         **indexers_kwargs: Any,
     ) -> T_DataArray:
         """Return a new DataArray whose data is given by each `n` value
-        along the specified dimension(s).
```
Comment (dcherian) [inline xarray/core/dataarray.py:1475]: This looks like an indentation problem.

**Comment by dcherian** [comment]
Here's how you can run the doctests locally in an environment created using
```
mamba env create -f ci/requirements/environment.yml
```

```
conda activate xarray-tests
pytest --doctest-modules xarray --ignore xarray/tests
```

**Comment by e-marshall** [comment]
> Here's how you can run the doctests locally in an environment created using
> 
> ```
> mamba env create -f ci/requirements/environment.yml
> ```
> 
> ```
> conda activate xarray-tests
> pytest --doctest-modules xarray --ignore xarray/tests
> ```

Okay, thanks ! I'll give that a try

**Comment by dcherian** [comment]
Thanks @e-marshall Doctests pass and the test failures should be fixed by #6667

**Review by andersy005 — APPROVED** [review]
Thanks very much @e-marshall for this addition and welcome to xarray 🥳🎉

**Pull request merged**